synfig-studio/src/gui/workarea.cpp

```diff
@@ -1754,15 +1753,16 @@ WorkArea::on_hruler_event(GdkEvent *event)
 	SYNFIG_EXCEPTION_GUARD_BEGIN()
 	switch(event->type) {
 	case GDK_BUTTON_PRESS:
+		from_ruler_event = true;
 		if (get_drag_mode() == DRAG_NONE && show_guides) {
 			set_drag_mode(DRAG_GUIDE);
-			curr_guide = get_guide_list_y().insert(get_guide_list_y().begin(), 0.0);
-			curr_guide_is_x = false;
+			curr_guide = get_guide_list().insert(get_guide_list().begin(),{synfig::Point(((1.0/2.0)*(drawing_area->get_window()->get_width())*get_pw())+ get_window_tl()[0], 0),
+																			   synfig::Angle::rad(0)});
```

Comment:
I don't know if we need to change, but FYI there is `synfig::Angle::zero()` :)

Reply:
I think it is fine this way. The angle in the `Guide` struct is of type `Angle::rad`. Angle::zero() returns an angle of type `Angle` .